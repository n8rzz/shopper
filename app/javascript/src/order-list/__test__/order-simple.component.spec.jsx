import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { OrderSimple } from '../order-simple.component';
import {
    completeItemsMock,
    locationMapMock,
} from '../__mock__/order-mocks';

describe('OrderSimple', () => {
    describe('.componentWillUnmount', () => {
        test('resets handler methods', () => {
            const component = shallow(<OrderSimple
                orderList={[]}
                locationMap={locationMapMock}
            />);

            component.unmount();

            expect(typeof component._onClickViewMoreHandler).toBe('undefined');
        });
    });

    describe('when passed an empty `orderList`', () => {
        test('renders correctly with valid props', () => {
            const component = shallow(<OrderSimple
                orderList={[]}
                locationMap={locationMapMock}
            />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('when passed a valid `orderList`', () => {
        let component = null;

        beforeEach(() => {
            component = shallow(<OrderSimple
                orderList={completeItemsMock}
                locationMap={locationMapMock}
            />);
        });

        afterEach(() => {
            component = null;
        });

        test('renders correctly with valid props', () => {
            expect(component).toMatchSnapshot();
        });

        test('renders the default number of items', () => {
            const listItems = component.find('.qa-orderSimple-item');

            expect(listItems.length).toBe(5);
        });
    });

    describe('when passed `limit` prop', () => {
        let component = null;

        beforeEach(() => {
            component = shallow(<OrderSimple
                orderList={completeItemsMock}
                limit={10}
                locationMap={locationMapMock}
            />);
        });

        afterEach(() => {
            component = null;
        });

        test('renders correctly', () => {
            expect(component).toMatchSnapshot();
        });

        test('renders the correct number of items', () => {
            const listItems = component.find('.qa-orderSimple-item');

            expect(listItems.length).toBe(completeItemsMock.length);
        });
    });

    describe('._onClickViewMore()', () => {
        let component = null;
        let eventStub = null;

        beforeEach(() => {
            eventStub = {
                preventDefault: sinon.spy(),
            };
            component = shallow(<OrderSimple
                orderList={completeItemsMock}
                limit={1}
                locationMap={locationMapMock}
            />);
        });

        afterEach(() => {
            component = null;
            eventStub = null;
        });

        test('calls event.preventDefault()', () => {
            const instance = component.instance();

            instance._onClickViewMore(eventStub);

            expect(eventStub.preventDefault.callCount).toBe(1);
        });

        describe('when #limit < props.orderItems.length', () => {
            test('should update #limit + LIMIT_INCREMENTOR', () => {
                const instance = component.instance();
                component.setState({ limit: 1 });

                instance._onClickViewMore(eventStub);

                expect(component.state('limit')).toBe(6);
            });
        });

        describe('when #limit > props.orderItems.length', () => {
            test('should set #limit to props.orderItems.length', () => {
                const instance = component.instance();
                component.setState({ limit: 10 });

                instance._onClickViewMore(eventStub);

                expect(component.state('limit')).toBe(completeItemsMock.length);
            });
        });
    });
});
