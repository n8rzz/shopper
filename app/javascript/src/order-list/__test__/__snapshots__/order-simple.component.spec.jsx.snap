// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSimple when passed \`limit\` prop renders correctly 1`] = `
<Fragment>
  <div
    className="vr_2"
  >
    <ul
      className="vlist"
    >
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-0"
      >
        <div>
          <a
            className="link"
            href="/orders/4"
          >
            Dickinson and Sons
          </a>
        </div>
        <div>
          complete - Dec 03, 2018
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-1"
      >
        <div>
          <a
            className="link"
            href="/orders/5"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-2"
      >
        <div>
          <a
            className="link"
            href="/orders/6"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-3"
      >
        <div>
          <a
            className="link"
            href="/orders/7"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-4"
      >
        <div>
          <a
            className="link"
            href="/orders/8"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-5"
      >
        <div>
          <a
            className="link"
            href="/orders/9"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
    </ul>
  </div>
</Fragment>
`;

exports[`OrderSimple when passed a valid \`orderList\` renders correctly with valid props 1`] = `
<Fragment>
  <div
    className="vr_2"
  >
    <ul
      className="vlist"
    >
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-0"
      >
        <div>
          <a
            className="link"
            href="/orders/4"
          >
            Dickinson and Sons
          </a>
        </div>
        <div>
          complete - Dec 03, 2018
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-1"
      >
        <div>
          <a
            className="link"
            href="/orders/5"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-2"
      >
        <div>
          <a
            className="link"
            href="/orders/6"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-3"
      >
        <div>
          <a
            className="link"
            href="/orders/7"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
      <li
        className="qa-orderSimple-item"
        key="orderSimple-item-4"
      >
        <div>
          <a
            className="link"
            href="/orders/8"
          >
            Erdman-Abbott
          </a>
        </div>
        <div>
          complete - May 24, 2020
        </div>
      </li>
    </ul>
  </div>
  <div>
    <button
      className="btn btn-wide mix-btn_inverse"
      onClick={[Function]}
      type="button"
    >
      View More
    </button>
  </div>
</Fragment>
`;

exports[`OrderSimple when passed an empty \`orderList\` renders correctly with valid props 1`] = `
<div>
  No Items to display
</div>
`;
